gems/aws-sdk-s3/lib/aws-sdk-s3/legacy_signer.rb

```diff
@@ -157,33 +156,12 @@ def signed_querystring_params(endpoint)
       end
 
       def uri_escape(s)
-
         #URI.escape(s)
 
-        # URI.escape is deprecated, replacing it with escape from webrick
-        # to squelch the massive number of warnings generated from Ruby.
-        # The following script was used to determine the differences
-        # between the various escape methods available. The webrick
-        # escape only had two differences and it is available in the
-        # standard lib.
-        #
-        #     (0..255).each {|c|
-        #       s = [c].pack("C")
-        #       e = [
-        #         CGI.escape(s),
-        #         ERB::Util.url_encode(s),
-        #         URI.encode_www_form_component(s),
-        #         WEBrick::HTTPUtils.escape_form(s),
-        #         WEBrick::HTTPUtils.escape(s),
-        #         URI.escape(s),
-        #       ]
-        #       next if e.uniq.length == 1
-        #       puts("%5s %5s %5s %5s %5s %5s %5s" % ([s.inspect] + e))
-        #     }
-        #
```

Comment:
I'm not against making this change. However, can we keep this script documentation? I would touch up the documentation to mention that webrick was used but dropped for ruby 3 support. This kind of historical documentation (especially something critical such as signing) is helpful.

Here is the updated script.

```
# (0..255).each {|c|
#   s = [c].pack("C")
#   e = [
#     CGI.escape(s),
#     ERB::Util.url_encode(s),
#     URI.encode_www_form_component(s),
#     WEBrick::HTTPUtils.escape_form(s),
#     WEBrick::HTTPUtils.escape(s),
#     URI.escape(s),
#     URI::DEFAULT_PARSER.escape(s)
#   ]
#   next if e.uniq.length == 1
#   puts("%5s %5s %5s %5s %5s %5s %5s %5s" % ([s.inspect] + e))
# }
```

Reply:
Of course. I assumed that since `URI::DEFAULT_PARSER.escape` is pretty much an alias it wasn't necessary anymore. But it's true that this comment was helpful for me so I added it back.